rest/blip_client_test.go

```diff
@@ -571,74 +573,194 @@ func getCollectionsForBLIP(_ testing.TB, rt *RestTester) []string {
 	return collections
 }
 
-func createBlipTesterClientOpts(tb testing.TB, rt *RestTester, opts *BlipTesterClientOpts) (client *BlipTesterClient, err error) {
+func NewBlipTesterClientOpts(opts *BlipTesterClientOpts) (client *BlipTesterClient) {
 	if opts == nil {
 		opts = &BlipTesterClientOpts{}
 	}
-	btc := BlipTesterClient{
+	client = &BlipTesterClient{
 		BlipTesterClientOpts: *opts,
-		rt:                   rt,
 	}
 
+	return client
+}
+
+func (btc *BlipTesterClient) Close() {
+	btc.pullReplication.Close()
+	btc.pushReplication.Close()
+	for _, collectionClient := range btc.collectionClients {
+		collectionClient.Close()
+	}
+	if btc.nonCollectionAwareClient != nil {
+		btc.nonCollectionAwareClient.Close()
+	}
+}
+
+func (btc *BlipTesterClient) RunDualClients(btc2 *BlipTesterClient, test func(t *testing.T), t *testing.T, rtConfig *RestTesterConfig) {
+	rt := NewRestTester(t, rtConfig)
+	btc.rt = rt
+	btc2.rt = rt
+	parentT := btc.rt.TB.(*testing.T)
+	parentT.Run("revTree", func(t *testing.T) {
+		// create blip replications on client 1
+		err := btc.createBlipTesterReplications()
+		require.NoError(t, err)
+		// create blip replications on client 2
+		err = btc2.createBlipTesterReplications()
+		require.NoError(t, err)
+
+		btc.rt.TB = t
+		defer func() { btc.rt.TB = parentT }()
+		test(t)
+	})
+	// if test is not wanting version vector subprotocol to be run, return before we start this subtest
+	if btc.SkipVersionVectorInitialization {
+		btc.rt.Close()
+		return
+	}
+	// teardown active replications on blip tester clients
+	btc.tearDownBlipClientReplications()
+	btc2.tearDownBlipClientReplications()
+	rt.Close()
+
+	// setup rest tester for new run
+	rt = NewRestTester(t, rtConfig)
+	defer rt.Close()
+	btc.rt = rt
+	btc2.rt = rt
+	parentT = btc.rt.TB.(*testing.T)
+	parentT.Run("versionVector", func(t *testing.T) {
+		// bump sub protocol version here before replication init
+		err := btc.createBlipTesterReplications()
+		require.NoError(t, err)
+		err = btc2.createBlipTesterReplications()
+		require.NoError(t, err)
+
+		btc.rt.TB = t
+		defer func() { btc.rt.TB = parentT }()
+		test(t)
```

Comment:
There's a lot of repetition in this function and in the other Run functions which I think can largely be reduced down into a common run function that take various parameters